### ISSUE TITLE
Utilize GPU computational capabilities

### ISSUE THREAD

**Issue by jonysy** [lychee-eng/miro#5]
#### Useful projects

* [collenchyma: Extendable HPC-Framework for CUDA, OpenCL and common CPU](https://github.com/autumnai/collenchyma)
* [arrayfire (rust wrapper)](http://arrayfire.org/arrayfire-rust/arrayfire/fn.median.html)
* [nvptx](https://github.com/japaric/nvptx) (rust-lang/rust#38789)

#### Articles

* [Computer Vision in ArrayFire - Part 1: Feature Detection](http://arrayfire.com/computer-vision-arrayfire-part-1/)
* [Fast Computer Vision with OpenCV and ArrayFire](http://blog.accelereyes.com/blog/2011/08/24/fast-computer-vision/)
* [OpenCV GPU Module](http://docs.opencv.org/2.4/modules/gpu/doc/introduction.html)

**Comment by akonneker** [comment]
Have you noticed this development?
https://github.com/rust-lang/rust/issues/38789

....and I just noticed it's the third on your list!
Whoops!

**Comment by jonysy** [comment]
@akonneker Yep, but I've not come across the issue you linked to. Looks like it only supports CUDA..? No?

In any case, I'm actually leaning towards building on top of collenchyma. My 'hard-fork' is located here: https://github.com/lychee-eng/parenchyma
